service/application/src/main/java/com/solace/maas/ep/event/management/agent/cli/EmaCommandLine.java

```diff
@@ -1,212 +1,85 @@
-        boolean completedSuccessfully = false;
-        if (dataCollectionFileService.findAllByScanId(scanId).size() >=
-                statuses.size() && !anyScanStatusesInDesiredState("FAILED", statuses)) {
-            log.info("Scan request [{}]: Scan completed successfully.", scanId);
-            completedSuccessfully = true;
-        }
-        return completedSuccessfully;
-    }
-
-    private void setScanType(MessagingServiceEntity messagingServiceEntity, ScanRequestBO scanRequestBO, String messagingServiceId) {
-        switch (messagingServiceEntity.getType().toLowerCase()) {
-            case "solace":
-                scanRequestBO.setScanTypes(List.of("SOLACE_ALL"));
-                break;
-            case "kafka":
-                messagingServiceDelegateService.getMessagingServicesRelations(messagingServiceId).stream()
-                        .findFirst()
-                        .ifPresentOrElse(messagingServiceEntity1 -> scanRequestBO.setScanTypes(List.of("KAFKA_ALL", "CONFLUENT_SCHEMA_REGISTRY_SCHEMA")),
-                                () -> scanRequestBO.setScanTypes(List.of("KAFKA_ALL")));
-                break;
-            case "confluent_schema_registry":
-                scanRequestBO.setScanTypes(List.of("CONFLUENT_SCHEMA_REGISTRY_SCHEMA"));
-                break;
-            default:
-                throw new RuntimeException("Unsupported messaging service type: " + messagingServiceEntity.getType());
-        }
-    }
-
-    public void waitForScanToComplete(String scanId) throws InterruptedException {
-        AtomicBoolean scanCompleted = new AtomicBoolean(false);
-
-        // The hard timeout is set to 30 minutes to prevent the scan from running forever
-        long hardTimeoutMillis = System.currentTimeMillis() + 30 * 60 * 1000;
-
-        // The initiated timout is set to 2 minutes to shortcut a scan that is stuck in the INITIATED state
-        // This typically occurs if there is an error in the broker URL or credentials
-        long initiatedTimeoutMillis = System.currentTimeMillis() + 2 * 60 * 1000;
-
-        while (System.currentTimeMillis() < hardTimeoutMillis && !scanCompleted.get()) {
-            Thread.sleep(2000);
-
-            List<ScanStatusEntity> statuses = scanStatusService.getScanStatuses(scanId);
-
-            if (scanCompleted(statuses)) {
-                scanCompleted.set(true);
-            } else if (System.currentTimeMillis() > initiatedTimeoutMillis && scanInitiated(statuses)) {
-                throw new RuntimeException("Scan is stuck in INITIATED state. Check broker URL and credentials.");
+        try {
+            if (args.length >= 1) {
+
+                String type = args[0];
+
+                if ("scan".equals(type)) {
+                    if (args.length > 2) {
+                        String messagingServiceId = args[1];
+                        String filePathAndName = args[2];
```

Comment:
In the case that there are more than 3 args provided, we are ignoring the other args. Is that a better practice than throwing a too many args error?

Reply:
Yes, good idea, I'll add a check for too many args.